packages/app-builder-lib/src/asar/asarUtil.ts

```diff
@@ -45,237 +43,114 @@ export class AsarPackager {
+  }
 
-      if (stat.isSymbolicLink()) {
-        const s = stat as any
-        this.fs.getOrCreateNode(pathInArchive).link = s.relativeLink
-        s.pathInArchive = pathInArchive
-        unpackedFileIndexSet.add(i)
-        continue
-      }
+  private async detectAndCopy(fileSets: ResolvedFileSet[], cancellationToken: CancellationToken) {
+    const taskManager = new AsyncTaskManager(cancellationToken)
+    const unpackedPaths = new Set<string>()
+    const copiedFiles = new Set<string>()
 
-      let fileParent = path.dirname(pathInArchive)
-      if (fileParent === ".") {
-        fileParent = ""
+    const matchUnpacker = (file: string, dest: string, stat: fs.Stats) => {
+      if (this.config.unpackPattern?.(file, stat)) {
+        log.debug({ file }, "unpacking")
+        unpackedPaths.add(dest)
+        return
       }
+    }
+    const writeFileOrSymlink = async (transformedData: string | Buffer | undefined, source: string, destination: string, stat: fs.Stats) => {
+      copiedFiles.add(destination)
 
-      if (currentDirPath !== fileParent) {
-        if (fileParent.startsWith("..")) {
-          throw new Error(`Internal error: path must not start with "..": ${fileParent}`)
-        }
-
-        currentDirPath = fileParent
-        currentDirNode = this.fs.getOrCreateNode(fileParent)
-        // do not check for root
-        if (fileParent !== "" && !currentDirNode.unpacked) {
-          if (unpackedDirs.has(fileParent)) {
-            currentDirNode.unpacked = true
-          } else {
-            await correctDirNodeUnpackedFlag(fileParent, currentDirNode)
-          }
-        }
+      // If transformed data, skip symlink logic
+      if (transformedData) {
+        return this.copyFileOrData(transformedData, source, destination, stat)
       }
 
-      const dirNode = currentDirNode!
-      const newData = transformedFiles == null ? undefined : transformedFiles.get(i)
-      const isUnpacked = dirNode.unpacked || (this.unpackPattern != null && this.unpackPattern(file, stat))
-      const integrity = newData === undefined ? await hashFile(file) : hashFileContents(newData)
-      this.fs.addFileNode(file, dirNode, newData == undefined ? stat.size : Buffer.byteLength(newData), isUnpacked, stat, integrity)
-      if (isUnpacked) {
-        if (!dirNode.unpacked && !dirToCreateForUnpackedFiles.has(fileParent)) {
-          dirToCreateForUnpackedFiles.add(fileParent)
-          await mkdir(path.join(this.unpackedDest, fileParent), { recursive: true })
+      const realPathFile = await fs.realpath(source)
+
+      if (source === realPathFile) {
+        return this.copyFileOrData(undefined, source, destination, stat)
+      } else {
+        const realPathRelative = path.relative(this.config.appDir, realPathFile)
+        const symlinkTarget = path.resolve(this.rootForAppFilesWithoutAsar, realPathRelative)
+        const isOutsidePackage = realPathRelative.startsWith("../")
+        if (isOutsidePackage) {
```

Comment:
Don't we have a unit test for workspaces? If not, I'll need to create one off of `master` and merge it prior so we can diff the support via snapshots of this migration PR

Reply:
Yes, there is. However, if you use the asarUnpack configuration inside a workspace, it will perform an additional filter on the dependent workspace packages in asarUtil. There's no unit test case for this situation. I've already added one, so you can merge the latest master and see if it passes.

https://github.com/beyondkmp/electron-builder/blob/a2871a1c81230b88dbb6d85937e39b88a5ea8091/test/src/HoistedNodeModuleTest.ts#L43-L57

Reply:
Merged in latest `master` and build still passes 🙂

Reply:
What about local dependencies in node_modules? 
`"local_module": "file:../../../local_module"`.

Should we add another test case like this?

Reply:
I'm not familiar with that notation actually, but yeah, that's a great callout for us to check. I don't think we have a unit test for it already. Would you be willing to create a test case for that off of `master` branch?

Reply:
Ok. This usage scenario should be quite common. I often use locally compiled electron modules for testing.

Reply:
https://github.com/electron-userland/electron-builder/pull/8586/files